sentry-core/src/profiling.rs

```diff
@@ -188,12 +187,15 @@ fn get_profile_from_report(
                 let instruction_addr = format!("{:p}", frame.ip as *mut core::ffi::c_void);
                 #[cfg(not(feature = "frame-pointer"))]
                 let instruction_addr = format!("{:p}", frame.ip());
-                *address_to_frame_id
-                    .entry(instruction_addr.clone())
-                    .or_insert_with(|| {
-                        frames.push(RustFrame { instruction_addr });
-                        (frames.len() - 1) as u32
-                    })
+                let rust_frame = RustFrame { instruction_addr };
+
+                address_to_frame_idx
+                    .get_index_of(&rust_frame)
+                    .unwrap_or_else(|| -> usize {
+                        address_to_frame_idx.insert(rust_frame);
+
+                        address_to_frame_idx.len() - 1
+                    }) as u32
```

Comment:
This can be as simple as `address_to_frame_idx.insert_full(rust_frame).0 as u32`.

Also, I would recommend to just save the raw `const *()` in the `IndexSet`. that would avoid allocating a string for all of these, and would also likely speed up the `IndexSet` operations as well. Once you write them out into the report, thats the time you can `format!()` stuff.